Create Release Notes for 1.0

Andriciuc opened the pull request:
Added initial files and modifications to include 1.0 release notes to the TOC and variables. Updates:

* updated the ToC names to make them in line with style guide
* updated variable with new release branch and fixed small release note name
* updated ## Release Highlights with topics:
* Added tickets
* Updated Upgrade considerations

Andriciuc committed:
added release notes v1.0

Added the draft release notes for 1.0, updated mkdocs and release notes md with paths

Andriciuc committed:
Update variables.yml

updated release date and number

Andriciuc committed:
Update release-notes-v1.0.md

small content update for upgrade considerations

Andriciuc committed:
small fixes to variables

version variable was funky, messing with mkdocs serve, so I fixed it

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into pg-tde-release-notes-1.0

Andriciuc committed:
Update release-notes.md

changed the title of the pdf to: Percona Transparent Database Encryption for PostgreSQL

Andriciuc committed:
Merge branch 'release-17.5.2' into pg-tde-release-notes-1.0

Andriciuc committed:
Update mkdocs.yml

updated the ToC names to make them in line with style guide

Andriciuc committed:
Update release-notes-v1.0.md

small fix to versions, instead of RC2 I mentioned all previous versions

Andriciuc committed:
small fixes

updated variable with new release branch and fixed small release note name

dutow commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,34 @@
+
+## Upgrade considerations
+
+`pg_tde` 1.0 is not backward compatible with previous `pg_tde` versions (like Release Candidate 2) due to significant changes in code. This means you cannot directly upgrade from one version to another. You must [uninstall](../how-to/uninstall.md) `pg_tde` first and then [install](../install.md) and configure the new 1.0 version.
```
Uninstall is not enough, use a clean installation

Andriciuc replied:
updated to: This means you cannot directly upgrade from one version to another. You must do **a clean installation** of `pg_tde`.

Andriciuc committed:
Merge branch 'release-17.5.2' into pg-tde-release-notes-1.0

Andriciuc committed:
Merge branch 'pg-tde-release-notes-1.0' of https://github.com/percona/postgres into pg-tde-release-notes-1.0

Andriciuc committed:
updated rn and variables

added new variable pg_tde version
updated rn to say we need a clean install of pg_tde not just uninstall

Andriciuc committed:
Merge branch 'release-17.5.2' into pg-tde-release-notes-1.0

Andriciuc committed:
Merge branch 'release-17.5.2' into pg-tde-release-notes-1.0

Andriciuc committed:
Update release-notes-v1.0.md

small update to Upgrade Considerations, updated Release highlights

Andriciuc committed:
new tickets and highlights

added new ticket items and highlights, fixed mkdocs TOC structure and update date

Andriciuc committed:
Merge branch 'release-17.5.2' into pg-tde-release-notes-1.0

Andriciuc committed:
Merge branch 'release-17.5.2' into pg-tde-release-notes-1.0

Andriciuc committed:
Merge branch 'release-17.5.2' into pg-tde-release-notes-1.0

Andriciuc committed:
Merge branch 'release-17.5.2' into pg-tde-release-notes-1.0

Andriciuc committed:
Merge branch 'release-17.5.2' into pg-tde-release-notes-1.0

jeltz commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,96 @@
+
+[Get Started](../install.md){.md-button}
+
+## Release Highlights
```
Not sure about this section at all. Shouldn't the main highltigt which goes first that it is now stable for encrypting realtion data?

Andriciuc replied:
Updated with relational data mention as the first highligh

jeltz commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,96 @@
+
+* **Streaming and logical replication compatibility**
+
+You can now use `pg_tde` in replication setups.
```
This has been true for a long time. Not sure what changed here.

Andriciuc replied:
removed all highlights and left only 1.0 and wal mention

Andriciuc committed:
Update release-notes-v1.0.md

updated with GA mention and stable for encrypting relational data

jeltz commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,96 @@
+
+* **Improved performance testing & automation**
+
+Bare-metal fuzz testing, performance benchmarking with CI/CD pipelines, and daily reporting integrations with Grafana and InfluxDB help ensure robustness and transparency.
```
Not sure if this even should be a highlight but if it is it should be written more from a user PoV and less from a developer PoV. The Grafana is not even public yet.

dutow replied:
And we are not using influxdb at all.

Andriciuc replied:
removed all highlights and left only 1.0 and wal mention

jeltz commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,96 @@
+
+* **Key management enhancements**
+
+Added SQL-level functions and CLI tools for rotating, validating, and managing encryption keys. Now more compliant with PostgreSQL standards.
```
No idea what this refers to.

Andriciuc replied:
removed all highlights except wal and 1.0 mention

AndersAstrand commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,100 @@
+
+* **Improved performance testing & automation**
+
+Bare-metal fuzz testing, performance benchmarking with CI/CD pipelines, and daily reporting integrations with Grafana and InfluxDB help ensure robustness and transparency.
```
What will "bare-metal" mean to a dba reading this? Why are you name-dropping tools we use? I don't even know if we actually use influxdb, but I don't see how this could be relevant to anyone reading this anyways.

What is it you want to communicate here really?

Did an LLM generate this?

Andriciuc replied:
removed all highlights and left only 1.0 and wal mention

AndersAstrand commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,100 @@
+
+* **Streaming and logical replication compatibility**
+
+You can now use `pg_tde` in replication setups.
```
"now" as opposed to when? This is the first release?

Andriciuc replied:
removed all highlights and left only 1.0 and wal mention

jeltz reviewed (changes requested):
I think we need to discuss on Slack what the highlights should contain and what version they should be since. Also some of the highlights look AI generated since they make no sense to me.

AndersAstrand commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,100 @@
+
+* **Key management enhancements**
+
+Added SQL-level functions and CLI tools for rotating, validating, and managing encryption keys. Now more compliant with PostgreSQL standards.
```
This oversells our CLI tools _a lot_. We have one CLI tool and it doesn't validate anything. Its only job is to be able to change configuration for a cluster that isn't starting.

Andriciuc replied:
removed all highlights except wal and 1.0 mention

AndersAstrand commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,100 @@
+
+## Known issues
+
+* The default `mlock` limit on Rocky Linux 8 for ARM64-based architectures equals the memory page size and is 64 Kb. This results in the child process with `pg_tde` failing to allocate another memory page because the max memory limit is reached by the parent process.
```
We don't use mlock() ourselves. Is there a problem with openssl on rocky linux?

dutow replied:
This was an internal CI issue when we were working on not swapping the keys.

jeltz replied:
We actually do :(.

```
	/* we don't want principal keys to end up paged to the swap */
	if (mlock(cacheEntry, sizeof(TDEPrincipalKey)) == -1)
		elog(ERROR, "could not mlock principal key cache entry: %m");
```

Andriciuc replied:
keeping mlock limitation for now

AndersAstrand commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,100 @@
+Contributor guides, PostgreSQL-style CLI, static analysis integration (Clang), and extensive refactoring improve maintainability and onboarding for new contributors.
+
+* **Security hardened**
+
```
I think we shouldn't focus on differences from the pre-release versions, but instead on what this version actually is.

Andriciuc replied:
removed all highlights except wal and 1.0 mention

dutow commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,100 @@
+
+* **Security hardened**
+
+Sensitive metadata redaction, improved error messages, proper SQLSTATE codes, and stricter key validation increase security and clarity.
```
We didn't add error codes, that ticket is still open. How did it get in the release notes?

Andriciuc replied:
I got the release notes list from the Jira dashboard, it seems it also takes open tickets, I did check before but this slipped through, thank you! Also, removed all highlights except wal and 1.0 mention.

Andriciuc committed:
Update release-notes-v1.0.md

removed highlights, removed tickets that were to do from list, added wal beta mention in highlights

Andriciuc committed:
Update release-notes-v1.0.md

small fix to GA header, removed extra **

dutow commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,68 @@
+
+## Changelog
+
+### New Features
```
The only new feature in this list is PG-1257, everything else is internal or documentation improvement.

Andriciuc replied:
removed all other internals, I wonder if docs should also be removed though, still will discuss this later.

dutow commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,68 @@
+
+### Improvements
+
+- [PG-1527](https://perconadev.atlassian.net/browse/PG-1527) – Added proper error codes for error messages in `pg_tde`
```
We didn't do this, this ticket was closed with a `duplicate` resolution.

Andriciuc replied:
removed

Andriciuc committed:
Update release-notes-v1.0.md

removed non-feature tickets and a duplicate improvement ticket

Andriciuc committed:
Merge branch 'release-17.5.2' into pg-tde-release-notes-1.0

Andriciuc committed:
Merge branch 'release-17.5.2' into pg-tde-release-notes-1.0

jeltz commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,61 @@
+
+`pg_tde` {{tdeversion}} is **not** backward compatible with previous `pg_tde` versions, like Release Candidate 2, due to significant changes in code. This means you **cannot** directly upgrade from one version to another. You must do **a clean installation** of `pg_tde`.
+
+## Known issues
```
This does feels way too prominent for my taste.

Andriciuc replied:
better to leave it in, it is important to hammer home this point even if it is 1.0

jeltz commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,61 @@
+### Improvements
+
+- [PG-1617](https://perconadev.atlassian.net/browse/PG-1617) – Removed relation key cache
+- [PG-1635](https://perconadev.atlassian.net/browse/PG-1635) – User-facing TDE functions now return void
```
We should probably mention @AndersAstrand 's API change here with creating keys being separated from setting them.

jeltz commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,61 @@
+- [PG-1592](https://perconadev.atlassian.net/browse/PG-1592) – Improve error logs when Server Key Info is requested without being created  
+- [PG-1593](https://perconadev.atlassian.net/browse/PG-1593) – Fixed runtime failures when invalid Vault tokens are allowed during key provider creation
+- [PG-1600](https://perconadev.atlassian.net/browse/PG-1600) – Fixed Postmaster error when dropping a table with an unavailable key provider  
+- [PG-1605](https://perconadev.atlassian.net/browse/PG-1605) – Fixed the creation of undeclared dependencies for `pg_tde_grant_database_key_management_to_role()`
```
This should instead be that we removed those functions. And as an improvement.

Andriciuc replied:
Rewrote as follows and added to improvements: Removed undeclared dependencies for `pg_tde_grant_database_key_management_to_role()`

jeltz commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,61 @@
+- [PG-1606](https://perconadev.atlassian.net/browse/PG-1606) – Fixed missing superuser check in role grant function leads to misleading errors  
+- [PG-1607](https://perconadev.atlassian.net/browse/PG-1607) – Made CA parameter optional when Vault server runs without SSL  
+- [PG-1608](https://perconadev.atlassian.net/browse/PG-1608) – Updated and fixed global key configuration parameters in documentation  
+- [PG-1613](https://perconadev.atlassian.net/browse/PG-1613) – Fixed the `pg_tde_change_key_provider` CLI tool
```
I think we could be slightly more sepcific here.

Andriciuc replied:
reworded to: - Tested and improved the `pg_tde_change_key_provider` CLI utility

jeltz commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,61 @@
+- [PG-1600](https://perconadev.atlassian.net/browse/PG-1600) – Fixed Postmaster error when dropping a table with an unavailable key provider  
+- [PG-1605](https://perconadev.atlassian.net/browse/PG-1605) – Fixed the creation of undeclared dependencies for `pg_tde_grant_database_key_management_to_role()`
+- [PG-1606](https://perconadev.atlassian.net/browse/PG-1606) – Fixed missing superuser check in role grant function leads to misleading errors  
+- [PG-1607](https://perconadev.atlassian.net/browse/PG-1607) – Made CA parameter optional when Vault server runs without SSL  
```
No, we did not do this. We just fixed the parameter order and documentation so it was more clear.

Andriciuc replied:
rewrote: - Improved CA parameter order and surrounding documentation for clearer interpretation.

Andriciuc committed:
Update release-notes-v1.0.md

janwieremjewicz commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v1.0.md:
```diff
@@ -0,0 +1,61 @@
+
+## Upgrade considerations
+
+`pg_tde` {{tdeversion}} is **not** backward compatible with previous `pg_tde` versions, like Release Candidate 2, due to significant changes in code. This means you **cannot** directly upgrade from one version to another. You must do **a clean installation** of `pg_tde`.
```
I think it would be good to ensure that since it's GA, from now on the releases are planned to be backwards compatible :)

jeltz reviewed (approved):
I am not happy with the reloease notes but if we want to release now let's go what we have.
